packages/core/src/expression-builders/base-sql-expression.ts

```diff
@@ -11,13 +11,20 @@ import type { Literal } from './literal.js';
 import type { Value } from './value.js';
 import type { Where } from './where.js';
 
+/**
+ * A symbol that can be used as the key for a static property on a BaseSqlExpression class to uniquely identify it.
+ */
+export const SQL_IDENTIFIER = Symbol('SQL_IDENTIFIER');
```

Comment:
I'd rather not add a runtime property for something that is purely typing

If you want to use a symbol, you could use the same approach we've used for branding other types, by using something like

```
declare const SQL_IDENTIFIER: unique symbol;
```

Reply:
Sure, I've updated this now.
I had to revert to a protected symbol as the type-only static value didn't seem to work.

Reply:
Woops approved a bit too quickly, they need `declare` too otherwise the property will really be added (but under the "undefined" key)

Reply:
Updated now